src/mame/technics/kn1500.cpp

```diff
@@ -0,0 +1,284 @@
+// license:GPL2+
+// copyright-holders:Felipe Sanches
+/******************************************************************************
+
+	Technics SX-KN1500 music keyboard driver
+
+******************************************************************************/
+
+#include "emu.h"
+#include "cpu/tlcs900/tmp95c061.h"
+#include "video/hd44780.h"
+#include "imagedev/floppy.h"
+#include "machine/upd765.h"
+#include "emupal.h"
+#include "screen.h"
+#include "speaker.h"
+
+
+namespace {
+
+class kn1500_state : public driver_device
+{
+public:
+	kn1500_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_maincpu(*this, "maincpu")
+		, m_fdc(*this, "fdc")
+		, m_lcdc(*this, "hd44780")
+	{ }
+
+	void kn1500(machine_config &config);
+
+private:
+	required_device<tmp95c061_device> m_maincpu;
+	required_device<upd765a_device> m_fdc;
+	required_device<hd44780_device> m_lcdc;
+
+	HD44780_PIXEL_UPDATE(kn1500_pixel_update);
+	void kn1500_palette(palette_device &palette) const;
+	void kn1500_tlcs900_porta_w(offs_t offset, uint8_t data);
+	void kn1500_tlcs900_portb_w(offs_t offset, uint8_t data);
+	void kn1500_tlcs900_port5_w(offs_t offset, uint8_t data);
+	void kn1500_tlcs900_port6_w(offs_t offset, uint8_t data);
+	void kn1500_tlcs900_port7_w(offs_t offset, uint8_t data);
```

Comment:
Prefix should be dropped, it's implicit from the class (i.e. `kn1500_state::kn1500_palette` -> `kn1500_state::palette_init`)